creator/studies/migrations/0031_patch_dataservice.py

```diff
@@ -0,0 +1,118 @@
+# Generated by Django 2.2.24 on 2021-11-02 18:23
+
+"""
+This migration patches the new attributes into the DataService Study
+entities using the spreadsheet found in this migrations folder.
+"""
+
+from django.conf import settings
+from django.db import migrations, models
+
+import pandas as pd
+import os
+import requests
+
+
+def patch_helper(kf_id, study_dict):
+    """
+    Helper function to patch the data in study_dict to the kf_id Study in
+    DataService.
+    """
+    # Patch DataService
+    try:
+        resp = requests.patch(
+            f"{settings.DATASERVICE_URL}/studies/{kf_id}",
+            json=study_dict,
+            timeout=settings.REQUESTS_TIMEOUT,
+            headers=settings.REQUESTS_HEADERS,
+        )
+    except requests.exceptions.RequestException as e:
+        raise Exception(f"Problem updating study {kf_id}: {e}")
+    
+    # Raise an error if it looks like study failed to update
+    if not resp.status_code == 200 or "results" not in resp.json():
+        error = resp.json()
+        if "_status" in error:
+            error = error["_status"]
+        if "message" in error:
+            error = error["message"]
+        raise Exception(f"Problem updating study {kf_id}: {error}")
+
+
+def patch_attrs(apps, schema_editor):
+    """
+    Patch the short_code, domain, and program attributes of Study objects
+    in the DataService using the provided spreadsheet.
+    """
+    # Do nothing if update DataService feature is not enabled
+    if not (
```

Comment:
Are you checking this bc these settings are not enabled in dev or something? Wondering why we need this check

Reply:
won't this migration error out anytime someone tries to run study-creator locally without a dataservice up without this check?

Reply:
Oh I'm sorry I forgot to thumbs up this. I wanted to double check the value of the feature flag in dev, qa to make sure it was set to true and it is. So I think this is a good check to have